### PR TITLE
My Home: Use the layout API results to render primary cards.

### PR DESCRIPTION
**Pull request by kwight**
In #40172 we wired in the Home layout API; this PR uses those results to determine which cards to render in the primary location. No visual or functional changes should happen between this PR and production.

**Testing Instructions**

* Load `https://wordpress.com/home/:site` in production, with a site that has not completed its checklist,
* Load the same site and URL on this branch.
* Verify both look and behave the same.
* Repeat the process with a site that has completed the checklist (and is therefore showing Quick Links).

### CODE REVIEW

**Comment by kwight** [comment]
I'm not sure I understand it, but might this also be a good use-case for `AsyncLoad />`?

**Comment by gwwar** [comment]
> I'm not sure I understand it, but might this also be a good use-case for AsyncLoad?

I think so. I'm not sure we have guidelines on the tradeoffs when using AsyncLoad. cc @Automattic/team-calypso @sgomes  

I would instinctively assume async load less important components so we can slim down section bundles (initial page load). So I would assume this lends itself to more things like secondary notices/nudge slots, but I do see that we're using this quite a bit in reader.

**Comment by gwwar** [comment]
Nice to see this coming together! Much tidier! One blocker I noticed before we land is the API is not catching established sites that were not eligible for the checklist. (eg was created before the site setup checklist was made).

Screen Shot 2020-03-27 at 4 02 57 PM

So two things we might want to try:
- [ ] Probably swap over all pieces at once if possible (this will make the PR a little larger). We can maybe feature flag if it gets too big
- [ ] Let's add a few unit tests if we can for these cases that we miss on the API side.

**Review by lcollette — APPROVED** [review]
LGTM!

**Review by gwwar — DISMISSED** [review]
We shouldn't land until the API accounts for those site cases

File: client/my-sites/customer-home/locations/primary/index.jsx
Comment (mmtr) [inline client/my-sites/customer-home/locations/primary/index.jsx]: Unrelated to this PR, but I wish we could get rid of that `checklistMode`. Maybe we can refactor the `ChecklistSiteSetup` component in a follow-up so we don't need to inject it. That will simplify logic here and all locations would be the same (meaning we could have a single `Location` component).

**Review by mmtr — COMMENTED** [review]
This looks really promising!

**Comment by sgomes** [comment]
> I think so. I'm not sure we have guidelines on the tradeoffs when using AsyncLoad. cc @Automattic/team-calypso @sgomes

I don't think there are any, and it would be hard to come up with some. We don't want the whole app lazily loading every component with spinners everywhere, but sometimes applying a bit of selective lazy loading can really help, particularly when there are large amounts of code behind a conditional.

Unfortunately, there isn't any easy rule I can provide, as most of the time it comes down to measuring both cases and seeing which one is best. If there is a rule to mention, it's the general rule when dealing with performance: avoid premature optimisation. Assume there is no performance issue (and prioritise readability/simplicity) until there's proof that there's a performance issue 🙂

**Comment by mmtr** [comment]
> One blocker I noticed before we land is the API is not catching established sites that were not eligible for the checklist

D41072-code should handle this.

File: client/my-sites/customer-home/locations/primary/index.jsx
Comment (gwwar) [reply client/my-sites/customer-home/locations/primary/index.jsx]: I'd maybe wait a bit on further refactors, @danhauk was exploring a number of variations/patterns that we might want to look into next. For example it could be that we move away from checklists all together.

Codewise I'm more of a fan of separate components, vs a really big smart component.

Comment (kwight) [reply client/my-sites/customer-home/locations/primary/index.jsx]: Let's not bother with it for now then 👍

**Comment by kwight** [comment]
I rebased, and verified behaviour for older sites with D41072-code (once the date gating is added to it). I'll leave off mucking with `<AsyncLoad />`, so this looks ready for a last check once the API fix is in. I'd rather not move it all over to the API at once; a number of tiny PRs for each location feels more prudent to me 🙃

File: client/my-sites/customer-home/locations/primary/index.jsx
Comment (mmtr) [inline client/my-sites/customer-home/locations/primary/index.jsx]: We'll need to update these identifies once D41010-code lands.

Comment (kwight) [reply client/my-sites/customer-home/locations/primary/index.jsx]: 4273d501800f5eec9a434a4b9d00227cd22eb086

**Review by gwwar — APPROVED** [review]
If folks are comfortable with turning this one one layout section at a time, feel free to :shipit: 

Screen Shot 2020-03-31 at 1 56 38 PM
Screen Shot 2020-03-31 at 1 57 22 PM

**Pull request merged**
